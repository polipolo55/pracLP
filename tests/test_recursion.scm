(define (factorial n)
  (if (<= n 1)
      1
      (* n (factorial (- n 1)))))

(define (fibonacci n)
  (if (< n 2)
      n
      (+ (fibonacci (- n 1)) (fibonacci (- n 2)))))


(define (suma-llista llista)
  (if (null? llista)
      0
      (+ (car llista) (suma-llista (cdr llista)))))

(define (main)
  (display "Prova de recursivitat amb factorial:")
  (newline)
  (display "Factorial de 5: ")
  (display (factorial 5))
  (newline)

  (display "Prova de recursivitat amb fibonacci:")
  (newline)
  (display "Fibonacci de 6: ")
  (display (fibonacci 6))
  (newline)

  (display "Prova de recursivitat amb suma-llista:")
  (newline)
  (display "Suma de '(1 2 3 4 5): ")
  (display (suma-llista '(1 2 3 4 5)))
  (newline)

  (display "Totes les proves de recursivitat han finalitzat!")
  (newline)
)