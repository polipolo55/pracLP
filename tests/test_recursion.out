Prova de recursivitat amb factorial:

Factorial de 5: 
120

Prova de recursivitat amb fibonacci:

Fibonacci de 6: 
8

Prova de recursivitat amb suma-llista:

Suma de '(1 2 3 4 5): 
15

Totes les proves de recursivitat han finalitzat!

